parser/internal/names/types.go

```diff
@@ -15,6 +15,61 @@ import (
 // otherwise needs to parse the package source to find it.
 type TrackedPackages map[string]string
 
+// Application presents all resolved names throughout the Application
+type Application map[*est.Package]*Resolution
+
+// PkgObjRef resolves the given ast.Node to a package path, object name and type arguments. In the case it fails to
+// resolve, it will return ("", "", nil).
+//
+// If the given node is a index or index list, then the type arguments will be extracted.
+//
+// If the given node is:
+// - an ident, then it is resolved to a package level deceleration in the same file as the ident.
+// - a selector, then it is resolved to an exported ident from the selected package.
+// - any other node will result in nothing being returned
+//
+// Note:
+// - This function accounts for scoping and variable shadowing. (i.e. it will not mistake a variable "foo" in a function
+//   for the package imported as "foo").
+// - This function does not validate the existence of the objName for objects referenced in other packages. As such it
+//   can return references for objects in Go Modules.
```

Comment:
can return? Should be cannot return, I believe?

Reply:
No it can.

For instance, you could import `encore.dev/cron` and use `cron.NewJob`. This function will resolve that and understand that there is an exported object called `NewJob` inside the `encore.dev/cron` package - however, it will not validate if `NewJob` actually exists or not.

Reply:
Ah, but it doesn't work for arbitrary packages, only for imports we've explicitly decided to track, correct?

Reply:
Yeah; true